ql/utilities/steppingiterator.hpp

```diff
@@ -34,38 +36,164 @@ namespace QuantLib {
         step_iterator(const step_iterator<OtherIterator>& i,
-                      typename boost::enable_if_convertible
-                      <OtherIterator,Iterator>::type* = 0)
-        : super_t(i.base()), step_(static_cast<BigInteger>(i.step())) {}
+                      typename std::enable_if<std::is_convertible
+                      <OtherIterator, Iterator>::value>::type* = 0)
+        : base_(i.base_), step_(static_cast<BigInteger>(i.step())) {}
+
         // inspector
         Size step() const { return static_cast<Size>(this->step_); }
-        // iterator adapter interface
+
+        /*! \deprecated This class no longer inherits from boost::iterator_adaptor`.
+                        Deprecated in version 1.31.
+        */
+        [[deprecated("Do not use the boost::iterator_adaptor interface.")]]
+        Iterator base() const { return base_; }
+
+        /*! \deprecated This class no longer inherits from boost::iterator_adaptor`.
+                        Deprecated in version 1.31.
+        */
+        [[deprecated("Do not use the boost::iterator_adaptor interface.")]]
         void increment() {
-            std::advance(this->base_reference(), step_);
+            std::advance(base_, step_);
         }
+
+        /*! \deprecated This class no longer inherits from boost::iterator_adaptor`.
+                        Deprecated in version 1.31.
+        */
+        [[deprecated("Do not use the boost::iterator_adaptor interface.")]]
         void decrement() {
-            std::advance(this->base_reference(), -step_);
+            std::advance(base_, -step_);
         }
-        void advance(typename super_t::difference_type n) {
-            this->base_reference() += n*(this->step_);
+
+        /*! \deprecated This class no longer inherits from boost::iterator_adaptor`.
+                        Deprecated in version 1.31.
+        */
+        [[deprecated("Do not use the boost::iterator_adaptor interface.")]]
+        void advance(typename std::iterator_traits<Iterator>::difference_type n) {
+            base_ += n*(this->step_);
         }
-        typename super_t::difference_type
+
+        /*! \deprecated This class no longer inherits from boost::iterator_adaptor`.
+                        Deprecated in version 1.31.
+        */
+        [[deprecated("Do not use the boost::iterator_adaptor interface.")]]
+        typename std::iterator_traits<Iterator>::difference_type
         distance_to(const step_iterator& i) const {
-            return (i.base()-this->base())/(this->step_);
+            return (i.base_ - base_) / step_;
+        }
+
+        step_iterator& operator=(const step_iterator& other) {
+            base_ = other.base_;
+            step_ = other.step_;
+            return *this;
+        }
+
+        step_iterator& operator++() {
+            base_ += step_;
+            return *this;
+        }
+
+        step_iterator operator++(int) {
+            base_ += step_;
+            return *this;
```

Comment:
Wrong semantics - the post-increment operatore should return a copy of the iterator before the increment

Reply:
My bad, I have fixed it now.